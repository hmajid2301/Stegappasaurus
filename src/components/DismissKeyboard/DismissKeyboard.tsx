import * as React from "react";
import { Keyboard, TouchableWithoutFeedback } from "react-native";

const DismissKeyboard = ({ children }: { children: React.ReactChild }) => (
  <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
    {children}
  </TouchableWithoutFeedback>
);
export default DismissKeyboard;
