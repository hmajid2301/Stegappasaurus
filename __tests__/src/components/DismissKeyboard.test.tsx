import { mount, shallow } from "enzyme";
import React from "react";
import { Keyboard, Text } from "react-native";

import DismissKeyboard from "~/components/DismissKeyboard";

describe("DimissKeyboard: Match snapshots", () => {
  test("1", () => {
    const component = shallow(
      <DismissKeyboard>
        <Text>Random Text</Text>
      </DismissKeyboard>
    );
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(
      <DismissKeyboard>
        <Text>Other Text</Text>
      </DismissKeyboard>
    );
    expect(component).toMatchSnapshot();
  });
});

describe("DimissKeyboard: OnPress Props", () => {
  test("TouchableWithoutFeedback", () => {
    const component = mount(
      <DismissKeyboard>
        <Text>Random Text</Text>
      </DismissKeyboard>
    );

    const spy = jest.spyOn(Keyboard, "dismiss");
    (component.find("TouchableWithoutFeedback").props() as any).onPress();
    expect(spy).toHaveBeenCalled();
  });
});
