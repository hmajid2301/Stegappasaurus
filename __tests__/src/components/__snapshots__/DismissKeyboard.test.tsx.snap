// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DimissKeyboard: Match snapshots 1 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <Text>
    Random Text
  </Text>
</TouchableWithoutFeedback>
`;

exports[`DimissKeyboard: Match snapshots 2 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <Text>
    Other Text
  </Text>
</TouchableWithoutFeedback>
`;
